go-chaos/cmd/cluster.go

```diff
@@ -45,11 +46,88 @@ func AddClusterCommands(rootCmd *cobra.Command, flags *Flags) {
 			return waitForChange(flags)
 		},
 	}
+	var scaleCommand = &cobra.Command{
+		Use:   "scale",
+		Short: "Scales the cluster to the given size",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return scaleCluster(flags)
+		},
+	}
 
 	rootCmd.AddCommand(clusterCommand)
 	clusterCommand.AddCommand(statusCommand)
 	clusterCommand.AddCommand(waitCommand)
 	waitCommand.Flags().IntVar(&flags.changeId, "changeId", 0, "The id of the change to wait for")
+	clusterCommand.AddCommand(scaleCommand)
+	scaleCommand.Flags().IntVar(&flags.brokers, "brokers", 0, "The amount of brokers to scale to")
+}
+
+func scaleCluster(flags *Flags) error {
+	k8Client, err := createK8ClientWithFlags(flags)
+	if err != nil {
+		panic(err)
+	}
+
+	err = k8Client.AwaitReadiness()
+	if err != nil {
+		return err
+	}
+
+	port := 9600
+	closePortForward := k8Client.MustGatewayPortForward(port, port)
+	defer closePortForward()
+
+	changeResponse, err := requestBrokerScaling(port, flags.brokers)
+	if err != nil {
+		return err
+	}
+	formatted, err := json.MarshalIndent(changeResponse, "", "  ")
+	if err != nil {
+		return err
+	}
+	fmt.Println(string(formatted))
+
+	_, err = k8Client.ScaleZeebeCluster(flags.brokers)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func requestBrokerScaling(port int, brokers int) (*ChangeResponse, error) {
+	brokerIds := make([]int32, brokers)
+	for i := 0; i < brokers; i++ {
+		brokerIds[i] = int32(i)
+	}
+	url := fmt.Sprintf("http://localhost:%d/actuator/cluster/brokers", port)
```

Comment:
❓ So I just give it a list of brokers Ids to scale? 🤔 What was the idea behind this? Could I also give a list which starts at 10 to 15? Why not having an endpoint where I can do like `inc` by one. Out of curiosity.

Reply:
Yes, that's correct. You can give it ids 10-15 if you want :+1: 

An inc by one endpoint could be added to the API. So far only have two endpoints for this, adding a broker with a given id and updating the list of brokers completely (this is what we use here)